Content.Server/Medical/HealthAnalyzerSystem.cs

```diff
@@ -59,10 +96,38 @@ private void OpenUserInterface(EntityUid user, EntityUid analyzer)
             if (!TryComp<ActorComponent>(user, out var actor) || !_uiSystem.TryGetUi(analyzer, HealthAnalyzerUiKey.Key, out var ui))
                 return;
 
-            _uiSystem.OpenUi(ui ,actor.PlayerSession);
+            _uiSystem.OpenUi(ui, actor.PlayerSession);
         }
 
-        public void UpdateScannedUser(EntityUid uid, EntityUid user, EntityUid? target, HealthAnalyzerComponent? healthAnalyzer)
+        private void OnDamageChanged(EntityUid uid, MobStateComponent component, DamageChangedEvent args)
+        {
+            var query = EntityQueryEnumerator<HealthAnalyzerComponent>();
+            while (query.MoveNext(out var healthAnalyserUid, out var healthAnalyserComp))
+            {
+                if (healthAnalyserComp.ScannedEntity != uid)
+                    continue;
+
+                //Get distance between health analyser and the scanned entity
+                var scannedEntityPosition = _transformSystem.GetMapCoordinates(uid);
+                var healthAnalyserPosition = _transformSystem.GetMapCoordinates(healthAnalyserUid);
+                var distance = (scannedEntityPosition.Position - healthAnalyserPosition.Position).Length();
+
+                //If they are on different maps, or the distance is greater than the Max Scan range
+                //Remove the scanned entity to prevent further updates
+                if (scannedEntityPosition.MapId != healthAnalyserPosition.MapId ||
+                    distance > healthAnalyserComp.MaxScanRange)
+                {
+                    healthAnalyserComp.ScannedEntity = EntityUid.Invalid;
+                    _cell.SetPowerCellDrawEnabled(uid, false);
+                }
+                else
+                {
+                    UpdateScannedUser(healthAnalyserUid, uid, healthAnalyserComp);
+                }
+            }
+        }
+
+        public void UpdateScannedUser(EntityUid uid, EntityUid? target, HealthAnalyzerComponent? healthAnalyzer)
```

Comment:
This is very expensive to subscribe to, you're better off using a separate component and handling cleanup properly.

Reply:
I have re-written this with a new component (HealthBeingAnalyzedComponent) 
and replaced the EntityQueryEnumerator with a HashSet in said component.
Hopefully that is a more optimised solution.